src/frontends/pytorch/src/input_model.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "input_model.hpp"
+
+#include "place.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+
+InputModel::InputModel(std::shared_ptr<TorchDecoder> model_decoder) : m_model_decoder(model_decoder) {
+    const auto& inputs = m_model_decoder->inputs();
+    for (size_t i = 0; i < inputs.size(); ++i) {
+        auto in_place = std::make_shared<pytorch::Place>(*this, inputs[i]);
+        for (const auto& name : in_place->get_names()) {
+            m_name_to_place.emplace(name, std::dynamic_pointer_cast<frontend::Place>(in_place));
+        }
+        auto type_any = simplified_type_interpret(m_model_decoder->get_input_type(i));
+        auto dtype = element::dynamic;
+        if (type_any.is<element::Type>()) {
+            dtype = type_any.as<element::Type>();
+        }
+        m_descriptors.emplace(inputs[i], PlaceDesc(dtype, m_model_decoder->get_input_shape(i)));
+    }
+    const auto& outputs = m_model_decoder->outputs();
+    for (size_t i = 0; i < outputs.size(); ++i) {
+        auto out_place = std::make_shared<pytorch::Place>(*this, outputs[i]);
+        for (const auto& name : out_place->get_names()) {
+            m_name_to_place.emplace(name, std::dynamic_pointer_cast<frontend::Place>(out_place));
+        }
+        auto type_any = simplified_type_interpret(m_model_decoder->get_output_type(i));
+        auto dtype = element::dynamic;
+        if (type_any.is<element::Type>()) {
+            dtype = type_any.as<element::Type>();
+        }
+        m_descriptors.emplace(outputs[i], PlaceDesc(dtype, m_model_decoder->get_output_shape(i)));
+    }
+}
+
+std::vector<ov::frontend::Place::Ptr> InputModel::get_inputs() const {
+    std::vector<ov::frontend::Place::Ptr> res;
+    for (const auto& input_idx : m_model_decoder->inputs()) {
+        auto place_it = m_name_to_place.find(std::to_string(input_idx));
+        FRONT_END_GENERAL_CHECK(place_it != m_name_to_place.end(), "Couldn't find Place for input.");
+        res.push_back(place_it->second);
+    }
+    return res;
+}
+
+std::vector<ov::frontend::Place::Ptr> InputModel::get_outputs() const {
+    std::vector<ov::frontend::Place::Ptr> res;
+    for (const auto& output_idx : m_model_decoder->outputs()) {
+        auto place_it = m_name_to_place.find(std::to_string(output_idx));
+        FRONT_END_GENERAL_CHECK(place_it != m_name_to_place.end(), "Couldn't find Place for output.");
+        res.push_back(place_it->second);
+    }
+    return res;
+}
+
+Place::Ptr InputModel::get_place_by_tensor_name(const std::string& tensor_name) const {
+    auto place_it = m_name_to_place.find(tensor_name);
+    FRONT_END_GENERAL_CHECK(place_it != m_name_to_place.end(), "Couldn't find Place for provided tensor name.");
```

Comment:
I think we should return `nullptr` according to the common Frontend API spec.:

```
    /// \brief Returns a tensor place by a tensor name following framework conventions, or
    /// nullptr if a tensor with this name doesn't exist.
    /// \param tensor_name Name of tensor
    /// \return Tensor place corresponding to specified tensor name or nullptr if not exists
    virtual Place::Ptr get_place_by_tensor_name(const std::string& tensor_name) const;

````